keras_cv/layers/preprocessing/random_resized_crop.py

```diff
@@ -0,0 +1,155 @@
+            to (0.08, 1.0).
+        interpolation: (Optional) A string specifying the sampling method for
+            resizing.
+        seed: (Optional) Integer. Used to create a random seed.
+    """
+
+    def __init__(
+        self,
+        target_size,
+        aspect_ratio_factor=(3.0 / 4.0, 4.0 / 3.0),
+        area_factor=(0.08, 1.0),
+        interpolation="bilinear",
+        seed=None,
+        **kwargs
+    ):
+        super().__init__(seed=seed, **kwargs)
+
+        self.target_size = target_size
+
+        if isinstance(aspect_ratio_factor, tuple):
+            min_aspect_ratio = min(aspect_ratio_factor)
+            max_aspect_ratio = max(aspect_ratio_factor)
+        elif isinstance(aspect_ratio_factor, (int, float)):
+            min_aspect_ratio = 3.0 / 4.0
+            max_aspect_ratio = aspect_ratio_factor
+        else:
+            min_aspect_ratio = 3.0 / 4.0
+            max_aspect_ratio = 4.0 / 3.0
+
+        self.aspect_ratio_factor = preprocessing.parse_factor(
+            aspect_ratio_factor,
+            min_value=min_aspect_ratio,
+            max_value=max_aspect_ratio,
+            param_name="aspect_ratio_factor",
+            seed=seed,
+        )
+        self.area_factor = preprocessing.parse_factor(
+            area_factor, param_name="area_factor", seed=seed
+        )
+
+        self.interpolation = interpolation
+        self.seed = seed
+
+        if area_factor == 0.0 and aspect_ratio_factor == 0.0:
+            warnings.warn(
+                "RandomResizedCrop received both `area_factor=0.0` and "
+                "`aspect_ratio_factor=0.0`. As a result, the layer will perform no "
+                "augmentation."
+            )
+
+    def get_random_transformation(self, image=None, label=None, bounding_box=None):
+        area_factor = self.area_factor()
+        aspect_ratio = self.aspect_ratio_factor()
+
+        new_height = tf.clip_by_value(
+            tf.sqrt(area_factor / aspect_ratio), 0.0, 1.0
+        )  # to avoid unwanted/unintuitive effects
+        new_width = tf.clip_by_value(tf.sqrt(area_factor * aspect_ratio), 0.0, 1.0)
+
+        height_offset = self._random_generator.random_uniform(
+            (),
+            minval=tf.minimum(0.0, 1.0 - new_height),
+            maxval=tf.maximum(0.0, 1.0 - new_height),
+            dtype=tf.float32,
+        )
+
+        width_offset = self._random_generator.random_uniform(
+            (),
+            minval=tf.minimum(0.0, 1.0 - new_width),
+            maxval=tf.maximum(0.0, 1.0 - new_width),
+            dtype=tf.float32,
+        )
+
+        y1 = height_offset
+        y2 = height_offset + new_height
+        x1 = width_offset
+        x2 = width_offset + new_width
+
+        return [[y1, x1, y2, x2]]
+
+    def augment_image(self, image, transformation):
+        image = tf.expand_dims(image, axis=0)
+        boxes = transformation
+
+        # See bit.ly/tf_crop_resize for more details
```

Comment:
Full links supported for now?

Reply:
Not sure about comments, so kept it as it is.

Reply:
@LukeWood

Reply:
good point, comments may not support full links yet.  I'll see if we can disable the lint check for this.